### PR TITLE
Fix permissions issue

### LINKED ISSUE #151: Dave Finley's permissions are not allowing sign ownership in signage.chapman.edu.
Trello Card: https://trello.com/c/gyy3vzlo/99-defect-dave-finley-s-permissions-are-not-allowing-sign-ownership-in-signage-chapman-edu

Podio Issue: https://podio.com/myorg-gfp97dxwpr/web-and-interactive-media/apps/build-jobs/items/1114

### PR DESCRIPTION
**Pull request by homeofmatt**
closes #151

### CODE REVIEW

File: Capfile [hunk 1]
```diff
@@ -26,5 +26,5 @@ require 'capistrano/passenger'
 require "whenever/capistrano"
 require "airbrussh/capistrano"
 
-# Load custom tasks from `lib/capistrano/tasks` if you have any defined
-Dir.glob('lib/capistrano/tasks/*.rake').each { |r| import r }
+# Loads custom tasks from `lib/capistrano/tasks' if you have any defined.
+Dir.glob('lib/capistrano/tasks/*.cap').each { |r| import r }
```
Comment (tatwell) [inline Capfile]: Did recommended practice change here re: cap task extensions?

File: README.md [hunk 2]
```diff
@@ -20,9 +20,7 @@ bundle install
 touch .env
 
 # Clone production data
-rake db:create db:migrate
-rake cascade:sync
-rake faculty:sync
+cap production db:clone
```
Comment (tatwell) [inline README.md]: Why is this a cap command? I associate `cap` with deployment. This doesn't seem related to deployment to me but with setting up developer's env.

File: config/deploy/production.rb [hunk 3]
```diff
@@ -1,61 +1,11 @@
...
 # ======================
-# Defines a single server with a list of roles and multiple properties.
-# You can define all roles on a single server, or split them:
-
-# server 'example.com', user: 'deploy', roles: %w{app db web}, my_property: :my_value
-# server 'example.com', user: 'deploy', roles: %w{app web}, other_property: :other_value
-# server 'db.example.com', user: 'deploy', roles: %w{db}
-server 'signage.chapman.edu', user: 'charles', roles: %w{app db web}
-
+server 'signage.chapman.edu', user: 'charles', roles: %w{app db web postgres}
```
Comment (tatwell) [inline config/deploy/production.rb]: You'll need to explain this new `postgres` role for me. Description in task below says:

> Clone the server database to your local machine

But this is being called for production here and staging below. So I'd be worried it was going to clone the server database somewhere everytime cap was run.

File: lib/capistrano/tasks/db.cap [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+      execute "pg_dump -F t -t users -t sponsors -t sign_users -t slide_users #{db_name} > #{remote_folder}#{backup_name}"
+    end
+    
+    sh "scp charles@#{hostname}:#{remote_folder}#{backup_name} #{local_folder}"
+    sh "RAILS_ENV=development bundle exec rake db:drop db:create db:migrate"
+    sh "RAILS_ENV=development bundle exec rake cascade:sync faculty:sync"
+    sh "pg_restore -O -d signage_development #{local_folder}#{backup_name}"
+    puts "Success"
+  end
+end
```
Comment (tatwell) [inline lib/capistrano/tasks/db.cap]: I still like the idea of this as a straight rake task. Making it cap task makes me worried it's going to be run automatically at times I don't want it to. A rake task indicates to me it will only be run when explicitly called.

File: README.md (hunk 2 above)
Comment (homeofmatt) [inline README.md]: I feel like capistrano can also be used for server automation tasks. We need to run the `pg_dump` task on the server and with a cap task we can specify which server we want to use.

**Comment by homeofmatt** [comment]
@tatwell okay I'm going to respond here. Here's an old article that I think still holds its ground on a good distinction on when to use Rake vs Capistrano.
http://ananelson.com/said/on/2007/12/30/remote-rake-tasks-with-capistrano/

File: config/deploy/production.rb [hunk 5]
```diff
@@ -1,61 +1,11 @@
...
-# group is considered to be the first unless any  hosts have the primary
-# property set. Specify the username and a domain or IP for the server.
-# Don't use `:all`, it's a meta role.
-
-# role :app, %w{deploy@example.com}, my_property: :my_value
-# role :web, %w{user1@primary.com user2@additional.com}, other_property: :other_value
-# role :db,  %w{deploy@example.com}
-
-
+# role :postgres, %w{charles@signage.chapman.edu}, no_release: true
```
Comment (homeofmatt) [inline config/deploy/production.rb]: Accident, I'll fix this

File: README.md (hunk 2 above)
Comment (mirabelrice) [inline README.md]: we also frequently use cap commands for tasks like these on our inside app.

File: config/deploy/production.rb (hunk 3 above)
Comment (homeofmatt) [inline config/deploy/production.rb]: It really doesn't matter. We can also use the `app`, `web`, or `db` role. I did it for consistency with the Inside app, where it was done because we have our database on a different server.

File: README.md (hunk 2 above)
Comment (tatwell) [inline README.md]: I worry this is going to create confusion in the future. `cap == deployment` in my mind. If you can point to something that suggests that this is a faulty or oversimplified association, I'll adjust.

I feel like the interface for this task should be:

    rake db:clone production

File: Capfile (hunk 1 above)
Comment (homeofmatt) [inline Capfile]: It literally makes no difference aside from making a clear distinction between what are capistrano tasks and what are rake tasks. Dependent on what's in this file, Capistrano will take either `.rake` or `.cap`
Personally I like separating Rake tasks and Cap tasks like this

File: README.md (hunk 2 above)
Comment (homeofmatt) [inline README.md]: Read the article below.

File: app/views/signs/show.html.erb [hunk 6]
```diff
@@ -48,7 +48,6 @@
     <div class="sign-slides-listing" <%= "id=js-sortable-slides" if can? :update, @sign %>>
       <%= render partial: 'sign_slide', collection: @sign.sign_slides.order(:order).eager_load(:slide) %>
     </div>
-    <%= link_to "+ New Slide", new_slide_path, class: 'btn green add-new-slide', id: 'add-slide-link' if can? :update, @sign %>
```
Comment (mirabelrice) [inline app/views/signs/show.html.erb]: So are we removing the option to add new slides here? was this the issue?

Comment (homeofmatt) [inline app/views/signs/show.html.erb]: Yes

File: README.md (hunk 2 above)
Comment (homeofmatt) [inline README.md]: @tatwell to also ease your worries, we have this functionality as a Capistrano task on Inside as well. Not once have we ever had an issue with any task being run at any point when it wasn't intended to be.

File: app/views/signs/show.html.erb (hunk 6 above)
Comment (tatwell) [inline app/views/signs/show.html.erb]: Thanks for the link. I still think we're dealing with different cases here. The article talks about tasks that still strike me as fundamentally related with deployment and are focused on observing/changing the state of a remote app server.

The point of the task here is to change the state of the local dev's environment and has nothing to do with deployment AFAICT. Making changes on a remote server (i.e. cloning a database) is just a side-effect.

So I still think a `rake` task makes more sense and better adheres to Principle of Least Surprise.

**Comment by mirabelrice** [comment]
Im a little unsure how some of these changes relate to the permissions issue, but the code looks good to me so 👍

File: README.md (hunk 2 above)
Comment (tatwell) [inline README.md]: If this follows an established pattern, then I'm more amenable to it. But I don't see any similar commands for inside, at least in the README.

Comment (tatwell) [inline README.md]: That said, "because we've always done it that way" is not always a great justification. I'd still like to know why we did it that way in the Inside app.

**Comment by homeofmatt** [comment]
@tatwell good points, let's make them rake tasks

**Comment by homeofmatt** [comment]
Should that be another card?

**Comment by tatwell** [comment]
An action item will work for now. You can always turn tasks into card.

Include a link to this PR.

**Comment by tatwell** [comment]
👍

**Comment by homeofmatt** [comment]
Sooo... merge this PR because it has the fix for the original issue and the clone task does work, just not how we want it.
Create an action item to migrate the clone tasks for both Inside and Signage to rake tasks
Yes?

**Comment by tatwell** [comment]
👍  I'd recommend separate tasks for Inside and Signage.

**Comment by homeofmatt** [comment]
Sounds good
